### PR TITLE
Fix Accept-Encoding handling to work correctly if only two compression schemes are available

### PR DESCRIPTION
**Pull request by palant**
## Description
This fixes `compression::get_preferred_encoding()` to consider all encodings listed in `Accept-Encoding` header and not only the top one. While at it, I replaced dynamic construction of available encodings on each call by a constant.

## Motivation and Context
As things are right now, `compression::get_preferred_encoding()` only handles the scenarios “one compression scheme enabled” and “all compression schemes enabled” correctly, with the former being handled as a special case. If for example only `compression-deflate` and `compression-gzip` features are enabled and the client requests `Accept-Encoding: br, gzip`, it will return `None` because Brotli isn’t enabled – yet it should have recognized gzip being enabled.

## How Has This Been Tested?
I’ve adjusted the integration tests to explicitly test for the scenario where the top listed encoding isn’t something we support (`identity`). While at it, I fixed a bug in this test – it failed if only some compression features were enabled but not all.

### CODE REVIEW

File: src/compression.rs [hunk 1]
```diff
@@ -228,43 +240,10 @@ pub fn get_preferred_encoding(headers: &HeaderMap<HeaderValue>) -> Option<Conten
     if let Some(ref accept_encoding) = headers.typed_get::<AcceptEncoding>() {
         tracing::trace!("request with accept-ecoding header: {:?}", accept_encoding);
```
Comment (joseluisq) [inline src/compression.rs]:
```suggestion
        tracing::trace!("request with accept-encoding header: {:?}", accept_encoding);
```

File: src/headers_ext/accept_encoding.rs [hunk 2]
```diff
@@ -69,11 +63,6 @@ mod tests {
         let val = HeaderValue::from_static("deflate, gzip;q=1.0, br;q=0.9");
```
Comment (joseluisq) [inline src/headers_ext/accept_encoding.rs]: Let's add `zstd` to the list just for completeness.

**Review by joseluisq — CHANGES_REQUESTED** [review]
Minor stuff to address and this bug below.

If for example, you use the SWS as a crate and enable just one compression feature like `default-features = false, features = ["compression-brotli"]` then you will get the following error.

```sh
error: variable does not need to be mutable
   --> static-web-server/src/handler.rs:276:25
    |
276 |                     let mut resp = None;
    |                         ----^^^^
    |                         |
    |                         help: remove this `mut`
```

This should be solved by shadowing the `resp` variable.

**Comment by palant** [comment]
> This should be solved by shadowing the `resp` variable.

Yes, I’ve already noticed that issue myself and addressed it in #368.

File: src/headers_ext/accept_encoding.rs (hunk 2 above)
Comment (palant) [reply src/headers_ext/accept_encoding.rs]: Done.

File: src/compression.rs (hunk 1 above)
Comment (palant) [reply src/compression.rs]: Done.

**Review by joseluisq — APPROVED** [review]
Thanks!

**Pull request merged**